crates/solver/src/order_balance_filter.rs

```diff
@@ -57,71 +59,68 @@ pub fn max_balance(orders: &[Order]) -> HashMap<Query, U256> {
+        if Some(order.metadata.owner) == ethflow_contract {
+            result.push(BalancedOrder::full(order));
+            continue;
+        }
+
+        let needed_balance = match max_transfer_out_amount(&order) {
+            // Should only ever happen if a partially fillable order has been filled completely
+            Ok(balance) if balance.is_zero() => continue,
+            Ok(balance) => balance,
+            Err(err) => {
+                // This should only happen if we read bogus order data from
+                // the database (either we allowed a bogus order to be
+                // created or we updated a good order incorrectly), so raise
+                // the alarm!
+                tracing::error!(
+                    ?err,
+                    ?order,
+                    "error computing order max transfer out amount"
+                );
                 continue;
             }
+        };
 
-            let needed_balance = match max_transfer_out_amount(&order) {
-                // Should only ever happen if a partially fillable order has been filled completely
-                Ok(balance) if balance.is_zero() => continue,
-                Ok(balance) => balance,
-                Err(err) => {
-                    // This should only happen if we read bogus order data from
-                    // the database (either we allowed a bogus order to be
-                    // created or we updated a good order incorrectly), so raise
-                    // the alarm!
-                    tracing::error!(
-                        ?err,
-                        ?order,
-                        "error computing order max transfer out amount"
-                    );
-                    continue;
-                }
-            };
-
-            if order.data.partially_fillable {
-                if remaining_balance == 0.into() {
-                    continue;
-                }
-                result.push(BalancedOrder {
-                    order,
-                    available_sell_token_balance: needed_balance.min(remaining_balance),
-                });
-                remaining_balance = remaining_balance.saturating_sub(needed_balance);
-            } else {
-                match remaining_balance.checked_sub(needed_balance) {
-                    Some(balance) => {
-                        result.push(BalancedOrder::full(order));
-                        remaining_balance = balance;
-                    }
-                    None => continue,
-                };
+        if order.data.partially_fillable {
+            let balance_for_order = std::cmp::min(needed_balance, *remaining_balance);
+            if balance_for_order == 0.into() {
+                continue;
             }
+            result.push(BalancedOrder {
+                order,
+                available_sell_token_balance: balance_for_order,
+            });
+            *remaining_balance -= balance_for_order;
```

Comment:
Not sure I'm understanding the PR description correctly. Doesn't it say that we **don't** want to assign balances for partially fillable orders. And that we need to know the **remaining** balance after assigning all the FOK orders?
These lines seem to do the opposite.

Reply:
This PR is a refactor without changing logic. I need this refactor for the next PR, which will make use of the remaining balances.